### PR TITLE
Add interface for DBSCAN

### PR DESCRIPTION
**Pull request by ablaom**
Replaces #11. Builds on https://github.com/JuliaAI/MLJBase.jl/pull/806, which was not available at that time.

Adds an interface for the DBSCAB model, implementing it as a `Static` model (no fit), as the model does not generalize to new data. Assignments are returned by `predict`; DBSCAN point types (core/boundary/noise) are accessed from the post-predict report. Detailed MLJ-compliant doc-string included.

```julia
using MLJBase
using MLJClusteringInterface

X, y  = make_moons(400, noise=0.09, rng=1)

model = DBSCAN(radius=0.13, min_cluster_size=5)
mach = machine(model)

# compute and output cluster assignments for observations in `X`:
yhat = predict(mach, X);

# compare with true labels:
compare = zip(yhat, y) |> collect;
julia> compare[1:10] # clusters align with classes
10-element Vector{Tuple{CategoricalArrays.CategoricalValue{Int64, UInt32}, CategoricalArrays.CategoricalValue{Int64, UInt32}}}:
 (1, 1)
 (2, 0)
 (2, 0)
 (1, 1)
 (2, 0)
 (2, 0)
 (2, 0)
 (2, 0)
 (2, 0)
 (2, 0)

# access post-predict report (`C` is core point, `N` is noise):
julia> unique(report(mach).point_types)
2-element Vector{Char}:
 'C': ASCII/Unicode U+0043 (category Lu: Letter, uppercase)
 'N': ASCII/Unicode U+004E (category Lu: Letter, uppercase)

# other report items:
julia> report(mach) |> keys
(:point_types, :nclusters, :cluster_labels, :clusters)
```

### CODE REVIEW

**Comment by ablaom** [comment]
@juliohm

File: test/runtests.jl [hunk 1]
```diff
@@ -28,25 +26,83 @@ X, y = @load_crabs
...
+    model = DBSCAN(radius=0.1)
+    mach = machine(model) # no training data
+    yhat = predict(mach, X)
+    @test yhat == yhat1
+    @test MLJBase.report(mach).point_types == report1.point_types
+    @test MLJBase.report(mach).nclusters == report1.nclusters
+
+end
+
+@testset "MLJ interface" begin
```
Comment (OkonSamuel) [inline test/runtests.jl:98]: Cool. The would avoid code duplication across packages.

**Review by OkonSamuel — APPROVED** [review]
LGTM!!!.

**Comment by ablaom** [comment]
Thanks @OkonSamuel for your review.

**Pull request merged**